src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -300,12 +353,36 @@ private void OnHostingUnhandledException(object arg)
 
             if (span != null)
             {
-                var exception = UnhandledExceptionExceptionFetcher.Fetch<Exception>(arg);
-
-                span.SetException(exception);
+                span.SetException(arg.As<UnhandledExceptionStruct>().Exception);
             }
         }
 
+        [DuckCopy]
+        public struct HttpRequestInStartStruct
+        {
+            public HttpContext HttpContext;
+        }
+
+        [DuckCopy]
+        public struct HttpRequestInStopStruct
+        {
+            public HttpContext HttpContext;
+        }
```

Comment:
Could we use the same struct for both cases?

Reply:
I have it separated for performance reasons, internally the ducktyping library uses a `CacheResult<T>` where the `T` is that struct type and caches the resulting proxy type for the target type. If we miss the cache it failback to a `ConcurrentDictionary` lookup.

Reply:
So it's un purpose to avoid the look-up.

Reply:
Thanks for clarifying!